### DISCUSSION TITLE
How often does minio automatically repair cluster data by default?

### DISCUSSION THREAD

**Discussion by wuyanfeiwork** [minio/minio#16127] [Q&A] [answered]
I deleted some files from the data directory on two of the 4 nodes in my minio cluster. I found that the cluster seemed to automatically recover on a regular basis, but I couldn't find any pattern.

So I want to know how often the cluster data is automatically repaired by default?

**Comment by klauspost** [comment, marked as answer]
It will continuously check objects to spread the load.

Objects are heal checked once every 512 cycles.

`mc admin scanner info ALIAS` will show how many versions has been heal scanned in the last minute.

**Reply by wuyanfeiwork** [reply]
> It will continuously check objects to spread the load.
> 
> Objects are heal checked once every 512 cycles.
> 
> `mc admin scanner info ALIAS` will show how many versions has been heal scanned in the last minute.

'mc admin scanner info ALIAS' execution error

![image](https://user-images.githubusercontent.com/104751498/204034481-f7da38d6-eb36-45b9-924a-e160485f00bd.png)

**Reply by klauspost** [reply]
Update.

**Comment by wuyanfeiwork** [comment]
@klauspost 
"Objects are heal checked once every 512 cycles."

How do I understand "512 cycles"? 

Please help me answer my questions, thank you very much.

**Reply by klauspost** [reply]
Objects are constantly scanned for ilm and usage, erc. A full tree scan is completed every 16 cycles.

Every 512 cycles all objects have been heal checked with different objects being selected every cycle.

**Reply by wuyanfeiwork** [reply]
Oh, I'm sorry, I still don't understand how long "512 cycles" is, and how long does it take?

I mainly want to know how long it will take the cluster to recover the directory data I deleted on one of the nodes, and whether the scan check time is closely related to the size of my data.

**Reply by klauspost** [reply]
>  how long does it take

Purely depends on your server speed and the number of objects.

The command above will show the time it took for the last 16 cycles. You can multiply that by 32 to get the full heal time.
